notificator/notificator_queue.py

```diff
@@ -80,14 +80,21 @@ async def _is_already_notified(self, rh_account_id, cve_id, event_type):
             self.notified_accounts[(rh_account_id, cve_id)] = new_events
         return event_type in new_events
 
-    @staticmethod
-    def _create_notif_events(cve):
+    def _create_notif_events(self, cve_id):
         """Create list of events for notifications"""
+        cve = self.conditions.get_cve(cve_id)
+        if not cve:
+            return []
         return [
             {
                 "metadata": {},
                 "payload": {
-                    "reported_cve": cve,
+                    "reported_cve": cve["cve"],
+                    "cvss_score": float(cve["cvss_score"]),
+                    "impact_id": cve["impact_id"],
+                    "known_exploit": cve["exploits"],
+                    "has_rule": cve["is_rule"],
+                    "publish_date": cve["public_date"].isoformat(),
```

Comment:
`cve["public_date"]` can be `None`, on some other places we use:

```
row["public_date"].isoformat() if row["public_date"] else None
```

Reply:
https://github.com/RedHatInsights/vulnerability-engine/blob/master/notificator/notificator.py#L40  the cve map is only built from CVEs which have public date non null, this is a way to detect that vmaas has already synced this cve, and we have metadata we need

Reply:
I was previously looking at the CVEs (especially the new ones and known exploits cves) all of them had the public date

Reply:
I see, then it's fine.